lib/puppet/functions/cd4pe_deployments/get_pipeline_trigger_events.rb

```diff
@@ -0,0 +1,41 @@
+require 'puppet_x/puppetlabs/cd4pe_client'
+require 'puppet_x/puppetlabs/cd4pe_function_result'
+
+# @summary Search for all the pipeline triggers that have occurred with a given SHA
+Puppet::Functions.create_function(:'cd4pe_deployments::get_pipeline_trigger_events') do
```

Comment:
Admittedly pedantic, (but hopefully not as bad as Robocop), could this be a `list` method? Otherwise, I have to parse for that "s" just to know it's doing something different.